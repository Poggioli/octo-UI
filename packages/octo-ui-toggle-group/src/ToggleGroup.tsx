import { forwardRef } from "@octo-ui/core";
import { StyledToggleGroupItem, StyledToggleGroupRoot } from "./styles";
import { ToggleGroupItemProps, ToggleGroupRootProps } from "./types";

const ToggleGroup = forwardRef<
  typeof StyledToggleGroupRoot,
  ToggleGroupRootProps
>(({ ...props }, forwardedRef) => {
  return <StyledToggleGroupRoot {...props} ref={forwardedRef} />;
});
ToggleGroup.toString = () => `.${StyledToggleGroupRoot.className}`;

// ========================================================================= //

const ToggleGroupItem = forwardRef<
  typeof StyledToggleGroupItem,
  ToggleGroupItemProps
>(({ ...props }, forwardedRef) => {
  return <StyledToggleGroupItem {...props} ref={forwardedRef} />;
});
ToggleGroupItem.toString = () => `.${StyledToggleGroupItem.className}`;

export { ToggleGroup, ToggleGroupItem };
export { StyledToggleGroupRoot, StyledToggleGroupItem };
