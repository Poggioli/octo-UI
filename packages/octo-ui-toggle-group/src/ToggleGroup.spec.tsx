import { render } from "@testing-library/react";
import { ToggleGroup, ToggleGroupItem } from "./ToggleGroup";

describe("<ToggleGroup />", () => {
  it(`GIVEN a <ToggleGroup />
      WHEN render is called
      THEN should render a <ToggleGroup />`, () => {
    const rendered = render(
      <ToggleGroup type="single" aria-label="Text alignment">
        {Array.from({ length: 3 }).map((_, index) => (
          <ToggleGroupItem
            key={index}
            value={`option-${index}`}
            aria-label={`Option ${index}`}
          >
            Option {index}
          </ToggleGroupItem>
        ))}
      </ToggleGroup>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <ToggleGroup />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ToggleGroup.toString()).toBe(".octo-UI-c-eWctCk");
  });

  it(`GIVEN a <ToggleGroupItem />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ToggleGroupItem.toString()).toBe(".octo-UI-c-iYXamd");
  });
});
