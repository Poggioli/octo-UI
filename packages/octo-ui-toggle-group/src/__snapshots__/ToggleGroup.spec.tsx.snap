// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleGroup /> GIVEN a <ToggleGroup />
      WHEN render is called
      THEN should render a <ToggleGroup /> 1`] = `
<div>
  <div
    aria-label="Text alignment"
    class="octo-UI-c-eWctCk octo-UI-c-eWctCk-etcbZa-color-unstyled octo-UI-c-eWctCk-fIEaQw-size-standard"
    dir="ltr"
    role="group"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-checked="false"
      aria-label="Option 0"
      class="octo-UI-c-iYXamd"
      data-radix-collection-item=""
      data-state="off"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Option 
      0
    </button>
    <button
      aria-checked="false"
      aria-label="Option 1"
      class="octo-UI-c-iYXamd"
      data-radix-collection-item=""
      data-state="off"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Option 
      1
    </button>
    <button
      aria-checked="false"
      aria-label="Option 2"
      class="octo-UI-c-iYXamd"
      data-radix-collection-item=""
      data-state="off"
      role="radio"
      tabindex="-1"
      type="button"
    >
      Option 
      2
    </button>
  </div>
</div>
`;
